### DISCUSSION TITLE
Content not showing in x-data condition

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#3950] [7. Bugs (archived)]
I may have found a bug: when a getter function is used in external x-data definition it does not resolve to true, but it does when used directly in the html.

**Comment by ekwoka** [comment]
Directives only work inside of an x-data tree

**Comment by simonferndriger** [comment]
Ah, sorry, forgot the surrounding `x-data`/`x-init`. However, I can't reproduce the case I have in my production code unfortunately. Well, I guess in that case it's probably best to close this... since it would be too much of a stretch to analyse production code.

**Reply by ekwoka** [reply]
Could link the site. I won't go deep debugging, but it may be a clear issue to the trained eye.

**Comment by simonferndriger** [comment]
Thanks, appreciate it! Unfortunately, I changed that part of the code so I don't have it in production myself anymore. But the thing was, when I used the get function inside the html with `x-show="getterFunction"` it worked, while when I was using the same condition within `x-data` (`return this.getterFunction`) it didn't. But since I can't reproduce it, I can't rule out a fault on my side on this...
